source/dubregistry/mongodb.d

```diff
@@ -6,27 +6,56 @@ import vibe.db.mongo.mongo : connectMongoDB;
 import vibe.db.mongo.settings : MongoClientSettings, MongoAuthMechanism, parseMongoDBUrl;
 import std.typecons : Nullable;
 
+/// Database name
 string databaseName = "vpmreg";
 private Nullable!MongoClientSettings _mongoSettings;
 
 @safe:
 
-MongoClientSettings mongoSettings() {
+/**
+Initializes mongoSettings if null
+
+Params:
+	allowUnauthenticated = whether to allow unauthenticated access
+
+Examples:
+	dub -- --allow-unauthenticated=true
+
+Returns:
+	MongoClientSettings
+*/
+MongoClientSettings mongoSettings(bool allowUnauthenticated = false)
+{
 	if (_mongoSettings.isNull)
 	{
 		import std.process : environment;
-		auto mongodbURI = environment.get("MONGODB_URI", environment.get("MONGO_URI", "mongodb://127.0.0.1"));
+
+		auto mongodbURI = environment.get("MONGODB_URI",
+				environment.get("MONGO_URI", "mongodb://127.0.0.1"));
 		logInfo("Found mongodbURI: %s", mongodbURI);
 		_mongoSettings = MongoClientSettings.init;
-		parseMongoDBUrl(_mongoSettings, mongodbURI);
-		_mongoSettings.authMechanism = MongoAuthMechanism.scramSHA1;
-		if (_mongoSettings.database.length > 0)
-			databaseName = _mongoSettings.database;
-		_mongoSettings.safe = true;
+
+		MongoClientSettings _settings = _mongoSettings.get;
+
+		parseMongoDBUrl(_settings, mongodbURI);
+
+		if (!allowUnauthenticated)
+			_settings.authMechanism = MongoAuthMechanism.scramSHA1;
+
+		if (_settings.database.length > 0)
+			databaseName = _settings.database;
+
+		_settings.safe = true;
+
+		_mongoSettings = _settings;
```

Comment:
Please keep in line with the current style, the empty line between each non-control statement is really not helping readability.

Reply:
I don't know if this was dfmt or not tbh. Sometimes it works well, sometimes less so